engine/apps/alerts/tasks/check_escalation_finished.py

```diff
@@ -32,6 +34,17 @@ def check_escalation_finished_task():
     )
 
```

Comment:
### Problem
I find calculating `estimate_escalation_finish_time` and use it in the query error prone.
For example if `estimate_escalation_finish_time` was never set, auditor will never evaluate this alert group and it will be missed.

### Possible Solution
I propose to use the following approach:
The desired state of the alert group is stored in the _escalation snapshot_ (see it attached below)
Escalation snapshot consists of:
* all escalation steps 
* current escalation step
* next escalation step ETA (this ETA is used by celery, which means we expect celery to run the next escalation step by the ETA)

Auditor retrieves all active alert groups and flags alert groups which have celery ETA < current time.

> ⚠️ We need to make sure that escalation snapshot was successfully set for the alert group, otherwise flag the alert group.

### Bonus task 😄 
We know all the escalation steps from snapshot which are expected to be completed and we can compare them to the alert group log records. Auditor can flag the alert group if it sees that some escalations steps were not completed.

### Example of the escalation snapshot
```json
{
  "channel_filter_snapshot": {
    "id": 1,
    "str_for_clients": "default",
    "notify_in_slack": true,
    "notify_in_telegram": false,
    "notification_backends": null
  },
  "escalation_chain_snapshot": {
    "id": 1,
    "name": "ildar"
  },
  "last_active_escalation_policy_order": 1,
  "escalation_policies_snapshots": [
    {
      "id": 1,
      "order": 0,
      "step": 13,
      "wait_delay": null,
      "notify_to_users_queue": [
        1
      ],
      "last_notified_user": null,
      "from_time": null,
      "to_time": null,
      "num_alerts_in_window": null,
      "num_minutes_in_window": null,
      "custom_button_trigger": null,
      "notify_schedule": null,
      "notify_to_group": null,
      "escalation_counter": 0,
      "passed_last_time": "2023-02-02T07:11:33.560022Z",
      "pause_escalation": false
    },
    {
      "id": 2,
      "order": 1,
      "step": 0,
      "wait_delay": "00:05:00",
      "notify_to_users_queue": [],
      "last_notified_user": null,
      "from_time": null,
      "to_time": null,
      "num_alerts_in_window": null,
      "num_minutes_in_window": null,
      "custom_button_trigger": null,
      "notify_schedule": null,
      "notify_to_group": null,
      "escalation_counter": 0,
      "passed_last_time": "2023-02-02T07:11:39.324953Z",
      "pause_escalation": false
    },
    {
      "id": 3,
      "order": 2,
      "step": 13,
      "wait_delay": null,
      "notify_to_users_queue": [
        1
      ],
      "last_notified_user": null,
      "from_time": null,
      "to_time": null,
      "num_alerts_in_window": null,
      "num_minutes_in_window": null,
      "custom_button_trigger": null,
      "notify_schedule": null,
      "notify_to_group": null,
      "escalation_counter": 0,
      "passed_last_time": null,
      "pause_escalation": false
    }
  ],
  "slack_channel_id": null,
  "pause_escalation": false,
  "next_step_eta": "2023-02-02T07:16:39.273824Z"
}
```

Reply:
@iskhakov lets discuss this. I think all of these points should be addressed now 😄